ffi/src/rank_info.rs

```diff
@@ -0,0 +1,56 @@
+use std::ffi::c_char;
+use std::ffi::c_float;
+use std::ffi::c_uchar;
+use std::ffi::c_uint;
+use std::ffi::c_int;
+use slippi_exi_device::SlippiEXIDevice;
+
+use crate::{with_returning};
+
+#[repr(C)]
+pub struct RustRankInfo {
+    pub rank: c_char,
+    pub rating_ordinal: c_float,
+    pub global_placing: c_uchar,
+    pub regional_placing: c_uchar,
+    pub rating_update_count: c_uint,
+    pub rating_change: c_float,
+    pub rank_change: c_int,
+}
+
+/// Fetches the rank information of the user currently logged in.
+#[unsafe(no_mangle)]
+pub extern "C" fn slprs_fetch_rank_info(exi_device_instance_ptr: usize) {
+    with_returning::<SlippiEXIDevice, _, _>(exi_device_instance_ptr, |device| {
+        device.rank_manager.fetch_rank();
+    })
+}
+
+/// Gets the most recently fetched rank information of the user currently logged in.
+#[unsafe(no_mangle)]
+pub extern "C" fn slprs_get_rank_info(exi_device_instance_ptr: usize) -> *mut RustRankInfo {

```

Comment:
Hmmm, are you sure you need to `Box` the return value here?

If you _do_ need to `Box` it, you'll need to also provide a cleanup method that recreates the rank info (e.g using `Box::from_raw` or whatever) and allows it to drop/dealloc properly on the Rust side.